test/lib/mock.js

```diff
@@ -76,7 +76,22 @@ module.exports = function mock(options) {
         });
         return s;
       });
+  }
 
+  function deleteAws() {
+    var bucket = config.get('img.uploads.dest.public');
+    var u = '/' + bucket + '?delete';
+    return nock('https://s3.amazonaws.com')
+      .post(u)
+      .reply(200, function(uri, body) {
+        var id = body.match(/<Key>([0-9a-f]{32})<\/Key>/)[1];
+        var s = through();
+        s.setEncoding = function() {};
+        local.delete(id).done(function() {
+          s.end();
+        });
+        return s;
+      });
```

Comment:
It took me a good long while to figure out how all the pieces fit together here, and I'm not that confident I understood it all.  Does this basically mock out requests to S3 and replace them with corresponding calls to the local image backend?  It may be worth a comment just to orient new readers.